### PR TITLE
File Storage Resource and Deployment

### PR DESCRIPTION
**Pull request by makenaoda9**
# Description
Implemented file storage schema and resource 
File Storage Resource Create to deploy a PowerScale with 'terraform apply' 
Updated client generation and changed impacted implementations

# GitHub Issues

| GitHub Issue # |
| -------------- |
| [TFM-2240](https://jira.cec.lab.emc.com/browse/TFM-2440)|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Tested against the mock server
Tested e2e deployment against a real backend 

- [x] Acceptance tests

### CODE REVIEW

**Comment by JimFromDell** [comment]
run e2e test

**Review by gibarrama — APPROVED** [review]

**Comment by JimFromDell** [comment]
run e2e test

**Comment by makenaoda9** [comment]
run e2e test

File: apex/helper/system_activation_helper.go [hunk 1]
```diff
@@ -214,36 +240,6 @@ func getClientActivationToken(ctx context.Context, activateSystemClientSystem mo
 	}
 }
 
-func getPowerScaleActivationToken(ctx context.Context, activateSystemClientSystem models.ActivationClientModel) (string, error) {
```
Comment (doriac11) [inline apex/helper/system_activation_helper.go:217]: Isnt this needed to activate PowerScale?

File: apex/helper/system_activation_helper.go [hunk 2]
```diff
@@ -144,12 +156,26 @@ func ActivateSystemClientSystem(ctx context.Context, clientAPI *apexClient.APICl
 // @return string current activation token. If it is an empty string we were not able to extract and user should try to post a new one
 func getApexActivationToken(ctx context.Context, clientAPI *apexClient.APIClient, systemID string, storageType apexClient.StorageProductEnum) string {
 	post := clientAPI.StorageSystemTokensAPI.StorageSystemTokensCreate(ctx)
+	accessToken := "test-dummy-token"
+
+	if storageType == apexClient.STORAGEPRODUCTENUM_POWERSCALE {
+		accessToken, _ = GetNewToken(ctx)
```
Comment (doriac11) [inline apex/helper/system_activation_helper.go:162]: This is for the Apex DI token, this code should not be used as part of the system_activation code.

File: apex/helper/token_helper.go [hunk 3]
```diff
@@ -108,6 +109,7 @@ func GetNewToken(ctx context.Context) (string, error) {
 	data.Set("subject_token_type", "urn:ietf:params:oauth:token-type:saml2")
 
 	response, err := sendGetTokenRequest(urlLink, method, header, data.Encode())
+	time.Sleep(15 * time.Second)
```
Comment (doriac11) [inline apex/helper/token_helper.go:112]: Is this needed? if not please remove

**Review by doriac11 — APPROVED** [review]

File: apex/helper/token_helper.go (hunk 3 above)
Comment (doriac11) [reply apex/helper/token_helper.go:112]: Talked in slack

File: apex/helper/system_activation_helper.go (hunk 2 above)
Comment (doriac11) [reply apex/helper/system_activation_helper.go:162]: Talked in slack

**Comment by doriac11** [comment]
run e2e test

**Review by shekhar-j — APPROVED** [review]

**Comment by JimFromDell** [comment]
run e2e test

**Review by JimFromDell — APPROVED** [review]

**Comment by makenaoda9** [comment]
run e2e test

**Pull request merged**